boring/src/ssl/mod.rs

```diff
@@ -703,37 +696,49 @@ impl SslContextBuilder {
         unsafe {
             init();
             let ctx = cvt_p(ffi::SSL_CTX_new(SslMethod::tls_with_buffer().as_ptr()))?;
+            ffi::SSL_CTX_set_raw_public_key_mode(ctx);
 
             Ok(SslContextBuilder::from_ptr(ctx, true))
         }
     }
 
-    /// Sets raw public key certificate in DER format.
-    pub fn set_rpk_certificate(&mut self, cert: &[u8]) -> Result<(), ErrorStack> {
-        unsafe {
-            cvt(ffi::SSL_CTX_set_server_raw_public_key_certificate(
-                self.as_ptr(),
-                cert.as_ptr(),
-                cert.len() as u32,
-            ))
-            .map(|_| ())
-        }
-    }
-
-    /// Sets RPK null chain private key.
-    pub fn set_null_chain_private_key<T>(&mut self, key: &PKeyRef<T>) -> Result<(), ErrorStack>
+    /// SSL_CTX_set_raw_public_key_and_key sets a raw public key (in
+    /// SubjectPublicKeyInfo format) and the corresponding private key for a TLS
+    /// server. References to the given `CRYPTO_BUFFER` and `EVP_PKEY` objects are
+    /// added as needed. Exactly one of `privkey` or `privkey_method` may be
+    /// non-NULL. `spki` must be non-NULL iff `privkey_method` is, otherwise the
+    /// public key will be generated from `privkey`. It returns one on success and
+    /// zero on error. Clients should install a custom verifier with
+    /// `SSL_CTX_set_custom_verify` or `SSL_set_custom_verify` to verify the server's
+    /// public key, which can be obtained via `SSL_get0_peer_pubkey` or
+    /// `SSL_get0_unparsed_peer_pubkey`. This is only supported for client connections
+    /// currently and only with TLSv1.3.
```

Comment:
Let's not just copy-paste the docs from boringssl, most of this stuff is only relevant in the context of C code.

Reply:
fixed